src/fields.jl

```diff
@@ -0,0 +1,15 @@
+import Base: getindex, setindex!
+
+function getindex(x::Stinger, field::StingerFields)
+    idx = Int(field)
+    basepointer = convert(Ptr{fieldtype(StingerGraph, idx)}, x.handle)
```

Comment:
You can remove this type introspection by specializing for the float case and int case separately. Although some benchmarks would prove if it is worth it.